src/fsharp/PrettyNaming.fs

```diff
@@ -445,16 +445,19 @@ module internal Microsoft.FSharp.Compiler.PrettyNaming
 
     let demangleProvidedTypeName (typeLogicalName:string) = 
         if typeLogicalName.Contains "," then 
-            let pieces = splitAroundQuotation typeLogicalName ',' 
-            if pieces.[1..] |> Array.forall (fun x -> tryDemangleStaticStringArg x |> Option.isSome) then
-                let argNamesAndValues = 
-                    pieces.[1..] |> Array.map (fun piece -> 
-                        match tryDemangleStaticStringArg piece with 
-                        | None -> raise (InvalidMangledStaticArg piece)
-                        | Some v -> v)
-                pieces.[0], argNamesAndValues
-            else 
-                typeLogicalName, [| |]
+            let pieces = splitAroundQuotation typeLogicalName ','
+            if pieces.Length = 2 && pieces.[1] = "" then
```

Comment:
Could tighten this up a bit with pattern deconstruction and avoiding duplicate Some/None check.

```fsharp
match pieces with
| [| x; "" |] -> x, [| |]
| _ ->
    let argNamesAndValues = pieces.[1..] |> Array.choose tryDemangleStaticStringArg
    if argNamesAndValues.Length = (pieces.Length - 1) then
        pieces.[0], argNamesAndValues
    else
        typeLogicalName, [| |]
```